kernel/ixgbe/src/queue_registers.rs

```diff
@@ -0,0 +1,89 @@
+//! Structs which provide access to the ixgbe device queue registers and store their backing memory pages.
+//! They implement the `RxQueueRegisters` and `TxQueueRegisters` traits which allows 
+//! the registers to be accessed through virtual NICs
+
+use super::regs::{RegistersRx, RegistersTx};
+use alloc::{
+    sync::Arc,
+    boxed::Box
+};
+use core::ops::{Deref, DerefMut};
+use core::mem::ManuallyDrop;
+use nic_queues::{RxQueueRegisters, TxQueueRegisters};
+use memory::MappedPages;
+
+
+/// Struct that stores a pointer to registers for one ixgbe receive queue 
+/// as well as a shared reference to the backing MappedPages where these registers are located.
```

Comment:
We tie the lifetime of the register struct with the backing mapped pages manually, rather than through an owned ref. Since the members of this struct can't be accessed outside of this crate, it's not possible to separate the two.